sdk/tables/azure-data-tables/azure/data/tables/_table_shared_access_signature.py

```diff
@@ -78,13 +81,27 @@ def generate_account_sas(
         resource_types,
         permission,
         expiry,
-        start=kwargs.pop("start", None),
-        ip_address_or_range=kwargs.pop("ip_address_or_range", None),
-        protocol=kwargs.pop("protocol", None),
+        start=start,
+        ip_address_or_range=ip_address_or_range,
+        protocol=protocol,
     )
 
 
-def generate_table_sas(credential: AzureNamedKeyCredential, table_name: str, **kwargs: Any) -> str:
+def generate_table_sas(
+    credential: AzureNamedKeyCredential,
+    table_name: str,
+    *,
+    permission: Optional[TableSasPermissions] = None,
```

Comment:
Should this support `str` as well?

Reply:
No, `TableSasPermissions` is not an Enum.

Reply:
After checking the implementation, we do support `str` for this parameter. So added `str` to signature and docstring.